### PR TITLE
Generalize data mungers, add roster update script

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description of Changes
This PR generalizes the individual data mungers into reusable modules, and then uses those modules to create a script which will add new officers and compute assignment information for all users. This will be incredibly helpful for continuing to update the data while SPD's roster changes (with hires/departures).

## Notes for Deployment

The following steps should be made in order to deploy this:
1. Download the officer CSV from production
1. Download the historic roster from `spd-lookup` (I just `git pull` and use the copy locally)
1. Use the CSV to generate the **three** CSVs to be uploaded (this also needs to be run from the `data-munging` directory): `./integrate_latest_roster.py [OO-OFFICER-CSV] [HISTORICAL-CSV] [OUTPUT-CSV]`
1. **[IMPORTANT]** Back up the database
1. **[IMPORTANT]** Truncate the assignments & jobs tables
1. Upload the files to the prod server
1. Use `just` to load the data: `just import --officers-csv /data/spd_2021-12-28_update__officers.csv --assignments-csv /data/spd_2021-12-28_update__assignments.csv --salaries-csv /data/spd_2021-12-28_update__salary_2021.csv "Seattle\ Police\ Department"`

## Screenshots (if appropriate)
image

## Tests and linting

 - [x] I have rebased my changes on `main`

 - [x] `just lint` passes

 - [x] `just test` passes

### CODE REVIEW

**Comment by AetherUnbound** [comment]
Getting some _very weird_ linting errors that are unrelated to the project. Investigating

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -9,13 +9,14 @@ jobs:
...
         with:
-          python-version: 3.8
-      - name: Install Dependencies
-        run: pip install pre-commit
-      - name: Run static checks
-        run: just lint
+          python-version: 3.9
+      - uses: pre-commit/action@v2.0.3
+        env:
+          SKIP: isort  # Fixme: this isn't working on GHA, "setuptools" not found
```
Comment (mpuckett159) [inline .github/workflows/pull_request.yml:17]: Screen Shot 2022-01-03 at 3 50 59 PM
FYI near as I can tell the isort test works just fine in GHAs. Not exactly sure what the issue was previously.

Comment (AetherUnbound) [reply .github/workflows/pull_request.yml:17]: Which run are you getting this from?

Comment (AetherUnbound) [reply .github/workflows/pull_request.yml]: Welp, guess whatever cache issue was happening is fixed now :upside_down_face:

**Pull request merged**